sklearn/tree/tests/test_tree.py

```diff
@@ -60,6 +60,13 @@
 
 from sklearn.utils import compute_sample_weight
 
+
+pytestmark = pytest.mark.filterwarnings(
```

Comment:
I do not think we need the per file filterwarning here. In this case, `test_max_features` and `test_max_features_deprecated` are the only ones that needs the decorator.